.github/workflows/smoketest-build.yml

```diff
@@ -0,0 +1,32 @@
+name: Smoke-Build
+
+on:
+  push:
+    branches:
+      - main
+  pull_request:
+
+jobs:
+
+  build:
+
+    strategy:
+      matrix:
+        os: [windows-latest, ubuntu-latest, macos-latest]
```

Comment:
I'm not a super huge fan of testing on MacOS for costs reasons. So we will have to be very careful or run it on demand.

Reply:
The `macos-latest` is a hosted (free) runner, To get the "full" benefit, it would be great to run on the `macos-latest-xlarge` image (as it builds on arm64) but that does indeed incur costs. Ultimately we need to wait for a free linux runner that is able to run on `arm64`.

Reply:
But, would also add that there are lots of warnings that are made visible by having this smoke test.

Reply:
Great then!